### ISSUE TITLE
running dump() fails when decode_responses=True

### ISSUE THREAD

**Issue by brianmaissy** [redis/redis-py#1254]
redis-py 3.3.6
python 3.6.9
Ubuntu 18.04
redis 4.0.9

I have a `Redis` instance that I created with `decode_responses=True`. When I run `.dump(key)` the decoder fails to decode the bytes response, which makes sense, because it isn't unicode.

```
  File "/home/brian/.local/lib/python3.6/site-packages/redis/client.py", line 3677, in execute
    return execute(conn, stack, raise_on_error)
  File "/home/brian/.local/lib/python3.6/site-packages/redis/client.py", line 3570, in _execute_transaction
    response = self.parse_response(connection, '_')
  File "/home/brian/.local/lib/python3.6/site-packages/redis/client.py", line 3636, in parse_response
    self, connection, command_name, **options)
  File "/home/brian/.local/lib/python3.6/site-packages/redis/client.py", line 853, in parse_response
    response = connection.read_response()
  File "/home/brian/.local/lib/python3.6/site-packages/redis/connection.py", line 687, in read_response
    response = self._parser.read_response()
  File "/home/brian/.local/lib/python3.6/site-packages/redis/connection.py", line 442, in read_response
    response = self._reader.gets()
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc3 in position 1: invalid continuation byte
```

**Comment by Stan1sl0s** [comment]
Hello,
same issue for sentinel part : 
```
self.master = self.sentinel.master_for(redis_sentinels_master, socket_timeout=socket_db_timeout, db=db,  encoding=u'utf-8', decode_responses=True)
```

```
self.master.get(myKey)
b'xyz'
```

**Comment by vcaboara** [comment]
I don't know why, but this seems to have broken for me as well.

**Comment by brianmaissy** [comment]
@chayim could it be that the fix in #1731 doesn't cover the use of non-decodable commands in a pipeline? (see the original stack trace above). In `_execute_transaction()` it looks like the options aren't passed through to `parse_response()`
